challenge.rb

```diff
@@ -1,26 +1,26 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+  return input.map{|input| input.capitalize}
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+  return input.select{|input| input == "dog"}
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+  return input.reject{|input| input == "dog"}
 end
 
 def count_the_animals(input)
-  #implement your solution here
+  return input.count
 end
 
 def fetch_the_first_two(input)
-  #implement your solution here
+  return input[0..1]
 end
 
 def fetch_CD_animals(input)
-  #implement your solution here
+  return input.keep_if{|input| input == "cat" or input == "dog"}
```

Comment:
The idea with this exercise is to catch all of the items that start with the letters 'c' or 'd'.  Also, there's a general preference for using `||` instead of `or`, because `or` doesn't behave quite as expected in some situations.